src/openrct2/ride/Vehicle.cpp

```diff
@@ -998,34 +998,29 @@ rct_vehicle_sound_params Vehicle::CreateSoundParam(uint16_t priority) const
  *
  *  rct2: 0x006BB9FF
  */
-void Vehicle::UpdateSoundParams() const
+void Vehicle::UpdateSoundParams(std::vector<rct_vehicle_sound_params>& vehicleSoundParamsList) const
 {
     if (!SoundCanPlay())
         return;
 
     uint16_t soundPriority = GetSoundPriority();
-    rct_vehicle_sound_params* soundParam;
     // Find a sound param of lower priority to use
-    for (soundParam = &gVehicleSoundParamsList[0];
-         soundParam < gVehicleSoundParamsListEnd && soundPriority <= soundParam->priority; soundParam++)
-        ;
+    auto soundParamIter = std::find_if(
+        vehicleSoundParamsList.begin(), vehicleSoundParamsList.end(),
+        [&soundPriority](rct_vehicle_sound_params param) { return soundPriority > param.priority; });
 
-    if (soundParam >= &gVehicleSoundParamsList[std::size(gVehicleSoundParamsList)])
+    if (soundParamIter == std::end(vehicleSoundParamsList))
         return;
 
-    if (gVehicleSoundParamsListEnd < &gVehicleSoundParamsList[std::size(gVehicleSoundParamsList)])
+    // Shift all sound params down one if using a free space
+    if (vehicleSoundParamsList.size() < vehicleSoundParamsList.capacity())
     {
-        gVehicleSoundParamsListEnd++;
+        vehicleSoundParamsList.insert(soundParamIter, CreateSoundParam(soundPriority));
```

Comment:
Is the order actually really important here? I think a normal push back will probably do, but I haven't tested that theory.

Reply:
I assumed it is, but didn't really experiment without, I imagined it would be something like "play only 14 sounds, if any of the other are of lower priority, they just get ignored to avoid becoming a mess of sounds"